lib/api_callable.js

```diff
@@ -190,19 +190,30 @@ function retryable(aFunc, retry, otherArgs) {
   return function retryingFunc(argument, callback) {
     var canceller;
     var timeoutId;
-    var error = new Error('Retry total timeout exceeded before any' +
-        'response was received');
     var now = new Date();
-    var deadline = now.getTime() + retry.backoffSettings.totalTimeoutMillis;
+    var deadline;
+    if (retry.backoffSettings.totalTimeoutMillis) {
+      deadline = now.getTime() + retry.backoffSettings.totalTimeoutMillis;
+    }
+    var retries = 0;
+    var maxRetries = retry.backoffSettings.maxRetries;
 
     /** Repeat the API call as long as necessary. */
     function repeat() {
       timeoutId = null;
-      if (now.getTime() >= deadline) {
-        callback(error);
+      if (deadline && now.getTime() >= deadline) {
+        callback(new Error('Retry total timeout exceeded before any' +
+            'response was received'));
+        return;
+      }
+
+      if (retries && retries >= maxRetries) {
+        callback(new Error('Number of retries exceeded before any response' +
+            'was resceived'));
```

Comment:
Changed to read `Exceeded maximum number of retries before any response was received`.